examples/dreamcast/libdream/keyboard/keyboard.c

```diff
@@ -46,25 +40,40 @@ void kb_test(void) {
             if(k > 0xff)
                 printf("Special key %04x\n", k);
 
-            if(k != 13) {
+            if(k != 10) {
                 bfont_draw(vram_s + y * 640 + x, 640, 0, k);
                 x += 12;
-            }
-            else {
-                x = 20;
-                y += 24;
+                if(x >= 620) {
+                    x = 20;
+                    y += 24;
+                }
             }
         }
+    }
+}
+
+static void on_key_event(maple_device_t *dev, kbd_key_t key,
+                         key_state_t state, kbd_mods_t mods,
+                         kbd_leds_t leds, void *user_data) {
+    kbd_state_t *kbd_state = (kbd_state_t *)dev->status;
 
-        thd_sleep(10);
+    printf("[%c%u] %c: %s\n",
+           'A' + dev->port, dev->unit,
+           kbd_key_to_ascii(key, kbd_state->region, mods, leds),
+           state.value == KEY_STATE_CHANGED_DOWN? "PRESSED" : "RELEASED");
+
+    if(key == KBD_KEY_ENTER && state.value == KEY_STATE_CHANGED_DOWN) {
+        x = 20;
+        y += 24;
     }
+
 }
 
 int main(int argc, char **argv) {
-    int x, y;
+    kbd_set_event_handler(on_key_event, NULL);
 
-    for(y = 0; y < 480; y++)
-        for(x = 0; x < 640; x++) {
+    for(int y = 0; y < 480; y++)
+        for(int x = 0; x < 640; x++) {
```

Comment:
As stated in my other comment, you're overshadowing static variables here (which had no reason to be static, and even less being one-character long).